test/db/migrate.test.js

```diff
@@ -246,3 +245,161 @@ describe(Support.getTestDialectTeaser('db:migrate'), () => {
+        .on('error', e => {
+          callback(e);
+        })
+        .on('data', data => {
+          result += data.toString();
+        })
+        .on('end', () => {
+          callback(null, result);
+        });
+    };
+
+    const runCli = function (cliArgs, callback) {
+      let result = '';
+      // avoid double callbacks
+      let done = callback;
+      return gulp
+        .src(Support.resolveSupportPath('tmp'))
+        .pipe(helpers.runCli(cliArgs))
+        .on('error', e => {
+          done(e);
+          done = () => {};
+        })
+        .on('data', data => {
+          result += data.toString();
+        })
+        .on('end', () => {
+          done(null, result);
+        });
+    };
+
+    it('--to', function (done) {
+      const self = this;
+      const migrationsPath = Support.resolveSupportPath('assets', 'migrations');
+      const migrations = fs.readdirSync(migrationsPath);
+      const createTriggers = migrations.filter(migration => migration.indexOf('createTestTableForTrigger') > -1);
+
+      prepare('--to ' + createTriggers, () => {
+        helpers.readTables(self.sequelize, tables => {
+          expect(tables).to.have.length(3);
+          expect(tables).to.contain('User');
+          expect(tables).to.contain('trigger_test');
+          done();
+        });
+      });
+    });
+
+    it('--to full migration in two parts', function (done) {
+      const self = this;
+      const migrationsPath = Support.resolveSupportPath('assets', 'migrations');
+      const migrations = fs.readdirSync(migrationsPath);
+      const createTriggers = migrations.filter(migration => migration.indexOf('createTestTableForTrigger') > -1);
+      const createPost = migrations.filter(migration => migration.indexOf('createPost') > -1);
+
+      prepare('--to ' + createTriggers, () => {
+        helpers.readTables(self.sequelize, tables => {
+          expect(tables).to.have.length(3);
+          expect(tables).to.contain('User');
+          expect(tables).to.contain('trigger_test');
+          runCli('db:migrate --to ' + createPost, () => {
+            helpers.readTables(self.sequelize, tables => {
+              expect(tables).to.have.length(4);
+              expect(tables).to.contain('Post');
+              done();
+            });
+          });
+        });
+      });
+    });
+
+    it('--to should exit with 0 when there are no migrations', function (done) {
+      const self = this;
+      const migrationsPath = Support.resolveSupportPath('assets', 'migrations');
+      const migrations = fs.readdirSync(migrationsPath);
+      const createTriggers = migrations.filter(migration => migration.indexOf('createTestTableForTrigger') > -1);
+
+      prepare('--to ' + createTriggers, () => {
+        helpers.readTables(self.sequelize, tables => {
+          expect(tables).to.have.length(3);
+          expect(tables).to.contain('User');
+          runCli('db:migrate --to ' + createTriggers, err => {
+            done(err);
```

Comment:
For proper verification, check for containment of `No migrations were executed, database schema was already up to date.` and error code 0 here